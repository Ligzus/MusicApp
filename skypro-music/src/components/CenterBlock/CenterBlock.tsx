import Filter from "../Filter/Filter";
import Playlist from "../Playlist/Playlist";
import Search from "../Search/Search";
import styles from "../CenterBlock/CenterBlock.module.css";
import { getTracks } from "@/api/tracks";
import { Track_Type } from "@/types/tracks";

const CenterBlock = async () => {
  let tracks: Track_Type[] = [];
  let error: string = "";

  try {
    tracks = await getTracks();
  } catch (err: unknown) {
    error =
      err instanceof Error
        ? "Ошибка при загрузке траков: " + err.message
        : "Неизвестная ошибка";
  }

  return (
    <div className={styles.mainCenterblock}>
      <Search />
      <h2 className={styles.centerblockH2}>Треки</h2>
      <Filter tracks={tracks} />
      <Playlist tracks={tracks} />
    </div>
  );
};

export default CenterBlock;
