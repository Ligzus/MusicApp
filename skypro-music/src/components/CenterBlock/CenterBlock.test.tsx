import { render, screen, waitFor } from "@testing-library/react";
import CenterBlock from "./CenterBlock";
import "@testing-library/jest-dom";

// Mock функции и данных
jest.mock("../../api/tracks", () => ({
  getTracks: jest.fn(() => Promise.resolve({ data: [] })),
}));

test("CenterBlock рендерится корректно", async () => {
  // Рендерим компонент и ожидаем его асинхронное поведение
  const { asFragment } = render(await CenterBlock());

  // Используем waitFor для ожидания завершения всех асинхронных операций
  await waitFor(() => {
    expect(screen.getByText("Треки")).toBeInTheDocument();
  });

  // Проверяем снимок
  expect(asFragment()).toMatchSnapshot();
});
