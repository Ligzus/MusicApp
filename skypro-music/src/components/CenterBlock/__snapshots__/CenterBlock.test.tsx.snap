// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterBlock рендерится корректно 1`] = `
<DocumentFragment>
  <div
    class="mainCenterblock"
  >
    <div
      class="centerblockSearch"
    >
      <svg
        class="searchSvg"
      >
        <use
          xlink:href="img/icon/sprite.svg#icon-search"
        />
      </svg>
      <input
        class="searchText"
        name="search"
        placeholder="Поиск"
        type="search"
        value=""
      />
    </div>
    <h2
      class="centerblockH2"
    >
      Треки
    </h2>
    <div
      class="centerblockFilter"
    >
      <div
        class="filterTitle"
      >
        Искать по:
      </div>
      <div>
        <div
          class="filterButton btnText"
        >
          исполнителю
        </div>
      </div>
      <div>
        <div
          class="filterButton btnText"
        >
          году
        </div>
      </div>
      <div>
        <div
          class="filterButton btnText"
        >
          жанру
        </div>
      </div>
    </div>
    <div
      class="centerblockContent"
    >
      <div
        class="contentTitle"
      >
        <div
          class="playlistTitleCol01"
        >
          Трек
        </div>
        <div
          class="playlistTitleCol02"
        >
          Исполнитель
        </div>
        <div
          class="playlistTitleCol03"
        >
          Альбом
        </div>
        <div
          class="playlistTitleCol04"
        >
          <svg
            class="playlistTitleSvg"
          >
            <use
              xlink:href="img/icon/sprite.svg#icon-watch"
            />
          </svg>
        </div>
      </div>
      <div
        class="contentPlaylist"
      />
    </div>
  </div>
</DocumentFragment>
`;
